### ISSUE TITLE
There is no protection of players' personal data!!

### ISSUE THREAD

**Issue by mrjnumber1** [rathena/rathena#3626]
* **rAthena Hash**: current
* **Client Date**: x
* **Server Mode**: x
* **Description of Issue**: 
  * Result: As it is now, players' passwords are stored as md5 hashes, or worse, plain text. This is quite an ugly thing, unfortunately.
  * Expected Result: It is irresponsible to continue with it as is. Using a slow or iterative COULD be an issue even though the login server itself doesn't do much. Something as simple as what's suggested on https://crackstation.net/hashing-security.htm would be good enough. In my opinion, pincodes and email addresses should be have an option to be stored behind a hash as well. i understand why emails may not need to be, as it's quite common for servers' dbs to leak and suddenly peoples' emails get farmed by scammers.
  * How to Reproduce: x
  * Official Information: x
* **Modifications that may affect results**: x

**Comment by vykimo** [comment]
I agree with you, at least propose different protection for passwords. And propose mail/pin protections too.

**Comment by Lemongrass3110** [comment]
@mrjnumber1 I totally share your concerns regarding all of this. But this would mean we have to drop some packet support for old (still used) login packets. And I guess I do not have to tell you that MD5 is a worthless approach.

Since server owners hate to use launchers that is a bit of a problem and launchers would only solve the login issue. I have not checked if it would be possible for pin code or mail yet.

**Comment by vykimo** [comment]
At least stockage in SQL should be encrypted for mails and pincodes.

**Comment by mrjnumber1** [comment]
we don't have to drop support for those old packets. we can still take the plain text passwords in, and then hash+salt those for storage in the db. of course, it's highly recommended to put a big annoying warning message about using plaintext passwords, since anyone who's on wireless will have their details easily sniffable (although it's highly unlikely that someone will do this for a ro server lol).
in fact, we may as well do the same even for the md5 sending login packets, too, unfortunately.

**Comment by Everade** [comment]
Would be great to put our focus a little bit more towards players data privacy and security.
I doubt that players expect even private servers to run with plaintext or badly hashed passwords these days.

@gepard-me brought this up **almost 6 years ago** on hercules and went into some details on handling options. It might help.
https://github.com/HerculesWS/Hercules/issues/74
